help command results in an exit_code 2 due to a breaking change on click

`click>=8.2.0` introduces this [change](https://github.com/pallets/click/pull/1489) which results in 3.10 and upwards to break on the `test_help` function like so:
```python
    def test_help():
        """Test that `genbadge` provides the right help"""
    
        runner = CliRunner()
        result = runner.invoke(genbadge_cmd, [], catch_exceptions=False)
    
>       assert result.exit_code == 0
E       assert 2 == 0
E        +  where 2 = <Result SystemExit(2)>.exit_code
```
To fix this you can use the following code to force the correct behaviour on all versions:
```python
@click.group(invoke_without_command=True)
@click.pass_context
def genbadge(ctx):
    """
    Commandline utility to generate badges.
    To get help on each command use:
        genbadge <cmd> --help
    """
    if ctx.invoked_subcommand is None:
        click.echo(ctx.get_help())
        ctx.exit(0)
```

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -1,5 +1,14 @@
 # Changelog
 
+### 1.1.3 - In Progress
+
+- Fix UserWarning for pkg_resources by migrating to importlib.resources. Fixes
+  [#44](https://github.com/smarie/python-genbadge/issues/44). PR
+  [#45](https://github.com/smarie/python-genbadge/pull/45) by [vonsteer](https://github.com/vonsteer).
+- Fix help exit_code issue when using newer versions of Python. Fixes 
+  [#46](https://github.com/smarie/python-genbadge/issues/46). PR
+  [#45](https://github.com/smarie/python-genbadge/pull/45) by [vonsteer](https://github.com/vonsteer).
+
 ### 1.1.2 - Bugfix and new packaging
 
 - Fix branch coverage rate comparison when used with --branch without actual branches. Fixes

--- a/setup.cfg
+++ b/setup.cfg
@@ -48,6 +48,7 @@ install_requires =
     requests
     # note: do not use double quotes in these, this triggers a weird bug in PyCharm in debug mode only
     pathlib2;python_version<'3.2'
+    importlib_resources;python_version<'3.9'
 tests_require =
     pytest
 ;     flake8

--- a/src/genbadge/main.py
+++ b/src/genbadge/main.py
@@ -35,8 +35,9 @@ SILENT_HELP = ("When this flag is active nothing will be written to stdout. Note
                "is used as the output file.")
 
 
-@click.group()
-def genbadge():
+@click.group(invoke_without_command=True)
+@click.pass_context
+def genbadge(ctx):
     """
     Commandline utility to generate badges.
     To get help on each command use:
@@ -44,7 +45,9 @@ def genbadge():
         genbadge <cmd> --help
 
     """
-    pass
+    if ctx.invoked_subcommand is None:
+        click.echo(ctx.get_help())
+        ctx.exit(0)
 
 
 @genbadge.command(name="tests",

--- a/src/genbadge/utils_badge.py
+++ b/src/genbadge/utils_badge.py
@@ -21,11 +21,11 @@ except ImportError:  # pragma: no cover
 try:
     # Python 3.9+
     from importlib.resources import files, as_file
-except ImportError:
+except ImportError:  # pragma: no cover
     try:
         # Python 3.7-3.8 (importlib_resources backport)
         from importlib_resources import files, as_file
-    except ImportError:
+    except ImportError:  # pragma: no cover
         files = None
         as_file = None
 

--- a/src/genbadge/utils_badge.py
+++ b/src/genbadge/utils_badge.py
@@ -17,7 +17,17 @@ try:
 except ImportError:  # pragma: no cover
     pass
 
-from pkg_resources import resource_string, resource_filename
+# Migration from pkg_resources to importlib.resources
+try:
+    # Python 3.9+
+    from importlib.resources import files, as_file
+except ImportError:
+    try:
+        # Python 3.7-3.8 (importlib_resources backport)
+        from importlib_resources import files, as_file
+    except ImportError:
+        files = None
+        as_file = None
 
 
 COLORS = {
@@ -185,16 +195,38 @@ def get_svg_badge(
     return template
 
 
+def _resource_string(package, resource_name):
+    """Fallback for importlib.resources in older Python versions."""
+    if files and as_file:
+        with as_file(files(package) / resource_name) as f:
+            return f.read_bytes()
+    else:
+        # Fallback to pkg_resources for older Python versions
+        from pkg_resources import resource_string
+
+        return resource_string(package, resource_name)
+
+def _resource_filename(package, resource_name):
+    """Fallback for importlib.resources in older Python versions."""
+    if files and as_file:
+        with as_file(files(package) / resource_name) as f:
+            return str(f)
+    else:
+        # Fallback to pkg_resources for older Python versions
+        from pkg_resources import resource_filename
+
+        return resource_filename(package, resource_name)
+
 def get_local_badge_template():
     """Reads the SVG file template fgrom the package resources"""
     template_path = "badge-template.svg"
     try:
-        template = resource_string("genbadge", template_path).decode('utf8')
+        template = _resource_string("genbadge", template_path).decode('utf8')
     except IOError:
         # error when running on python 2 inside the CliInvoker from click with a change of os.cwd.
         import genbadge
         reload(genbadge)  # noqa
-        template = resource_string("genbadge", template_path).decode('utf8')
+        template = _resource_string("genbadge", template_path).decode('utf8')
 
     return template
 
@@ -221,12 +253,12 @@ def preferred_width_of(txt, font_name, font_size):
         font = ImageFont.truetype(font=font_file, size=font_size)
     except (IOError if sys.version_info < (3,) else OSError):
         # Font not found: use the embedded font file from the package
-        font_path = resource_filename("genbadge", font_file)
+        font_path = _resource_filename("genbadge", font_file)
         if not os.path.exists(font_path):
             # error when running on python 2 inside the CliInvoker from click with a change of os.cwd.
             import genbadge
             reload(genbadge)  # noqa
-            font_path = resource_filename("genbadge", font_file)
+            font_path = _resource_filename("genbadge", font_file)
 
         font = ImageFont.truetype(font=font_path, size=font_size)
 

